nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestColdStorage.java

```diff
@@ -108,6 +110,66 @@ public void shouldFailWhenMovingToColdStorageDocumentWithoutContent() {
         }
     }
 
+    @Test
+    public void shouldRetrieveDocumentBlobFromColdStorage() throws IOException {
+        DocumentModel documentModel = createDocument(true);
+
+        // move the blob to cold storage
+        ColdStorageHelper.moveContentToColdStorage(session, documentModel.getRef());
+
+        // retrieve, which means initiate a request to restore the blob from cold storage
+        documentModel = ColdStorageHelper.retrieveContentFromColdStorage(session, documentModel.getRef(),
+                NUMBER_OF_DAYS_OF_AVAILABILITY);
+        transactionalFeature.nextTransaction();
+        documentModel = session.getDocument(documentModel.getRef());
+
+        assertTrue((Boolean) documentModel.getPropertyValue(ColdStorageHelper.COLD_STORAGE_BEING_RETRIEVED_PROPERTY));
+
+        // check that `file:content` still contains the main content blob
+        checkBlobContent(documentModel, ColdStorageHelper.FILE_CONTENT_PROPERTY, FILE_CONTENT);
+
+        // check that `coldstorage:coldContent` still contains the original file content
+        checkBlobContent(documentModel, ColdStorageHelper.COLD_STORAGE_CONTENT_PROPERTY, FILE_CONTENT);
```

Comment:
I would remove these 2 assertions since the retrieval only updates the boolean field.